File: problemas/strings/1632/1632.go
package main

import (
	"fmt"
	"strings"
)

func variacoes(letra string) int {
	if letra == "A" || letra == "E" || letra == "I" || letra == "O" || letra == "S" {
		return 3
	} else {
		return 2
	}
}

func main() {
	var N int
	var senha string

	fmt.Scanf("%d\n", &N)
	for i := 0; i < N; i++ {
		fmt.Scanf("%s\n", &senha)
		senha = strings.ToUpper(senha)

		resposta := 1
		for j := 0; j < len(senha); j++ {
			resposta *= variacoes(string(senha[j]))
		}
		fmt.Printf("%d\n", resposta)
	}
}

File: problemas/strings/1632/1632.r
input <- file('stdin', 'r')

variacoes <- function(letra) {
    if (letra == 'A' || letra == 'E' || letra == 'I' || letra == 'O' || letra == 'S') {
        return(3)
    } else {
        return(2)
    }
}

n <- as.integer(readLines(input, n=1))
for (i in 1:n) {
    senha <- toupper(readLines(input, n=1))

    resposta <- 1
    for (j in 1:nchar(senha)) {
        resposta <- resposta * variacoes(substr(senha, j, j))
    }
    write(paste(resposta), '')
}